ui/src/config/section/user.js

```diff
@@ -96,6 +96,19 @@ export default {
           record.state === 'enabled'
       }
     },
+    {
+      api: 'lockUser',
+      icon: 'LockOutlined',
+      label: 'label.action.lock.user',
+      dataView: true,
+      popup: true,
+      show: (record, store) => {
+        return ['Admin', 'DomainAdmin'].includes(store.userInfo.roletype) && !record.isdefault &&
+          !(record.domain === 'ROOT' && record.account === 'admin' && record.accounttype === 1) &&
+          record.state === 'enabled'
+      },
+      component: shallowRef(defineAsyncComponent(() => import('@/views/iam/LockUser.vue')))
```

Comment:
@winterhazel I'm not sure but isn't there a mechanism already to show a confirm dialog with custom text? I guess that's all this new vue component is doing?

Reply:
There is. I created this component because I thought it would be nice to include the user's name in the warning and success messages so that operators do not accidentally lock the wrong user and so that they get a confirmation showing which user was locked, which was not possible with the existing mechanism since it does not provide the option to replace placeholders.

Reply:
That makes sense @winterhazel , but on second though, should we maybe extend the existing component instead? just a though, no :-1:

Reply:
@DaanHoogland this seems like the optimal solution, as it would allow easily doing the same thing in other parts of the UI. Since this feature would prove useful to this PR, I will try implementing it here.

Reply:
@DaanHoogland @shwstppr I have made some changes in this PR to allow the replacement of placeholders in `message`s and `successMessage`s of confirmation dialogs, and updated the description. Could you guys review again? Thanks.